app/routes.py

```diff
@@ -110,26 +113,43 @@ def categoryHandler(id):
         except Exception as e:
             return str(e), 500
 
-
-    else: 
+    else:
         if (Categories.findById(id) is None):
             return 'fail', 204
         Categories.delete(id)
         return "success", 200
 
 ################################################################################
-#matches 
+# matches
 ################################################################################
-    
+
+
 @cross_origin()
-@app.route("/api/matches", methods = ["GET"])
+@app.route("/api/matches", methods=["GET"])
 def getMatches():
-    app.logger.debug('This endpoint was hit!')
+    # app.logger.debug('This endpoint was hit!')
     matches = Matches.query.all()
     return json.dumps([m.serialize() for m in matches])
 
-@cross_origin()
-@app.route("/api/match/<id>", methods = ["DELETE", "GET", "PUT"])
+
+@cross_origin
+@app.route('/api/matches/<page>', methods=['GET'])
+def getMatchPage(page):
+    # app.logger.debug('hit match pagination')
+    matches = Matches.query.all()
+    matches = matches[::-1]
+    matchPage = matches[(int(page)-1)*10:int(page)*10]
+    return json.dumps([m.serialize() for m in matchPage])
+
+
+@app.route('/api/matches/count', methods=['GET'])
+def getMatchesCount():
+    matches = Matches.query.all()
+    return str(int(len(matches)))
```

Comment:
Looks good! 

As far as I can see, the getMatchesCount gets all of the matches, which is already happening in getMatchPage call. Is there a way that we can get the length in the getMatchPage query? 

This implementation works for now, but what happens if we want to change the length of results per page as a request in the future or we want to enhance it to also query based on date? Could we use REST API parameters to pass this information?

Reply:
ill take a loop at the resposniveness, and ig i can return metadata about the matches? but that would have to be in the header....
regarding the length of results, that's just a magic number that i can also turn into a variable for easy change. idk how to do REST API parameters tho

Reply:
Something that can work is the make /api/matches return a json object like this:
```
{ 
    metadata: {
        page: int
        max_page: int
        number_matches: int
        other_useful_data: 
    }
    records: [
        match1,
        match2, 
        etc
    ]
}
```
For parameters, in flask you can access the URL parameters by using `request.args` 
https://flask.palletsprojects.com/en/2.2.x/api/#flask.Request.args

So in the end, an example request might look like `base_url/api/matches?page=2&results_per_page=20` or something,
Example here: https://www.botify.com/learn/basics/what-are-url-parameters